lib/cylc/task_pool.py

```diff
@@ -795,63 +795,71 @@ def warn_stop_orphans(self):
                     self.log.warning("%s: incomplete task event handler %s" % (
                         itask.identity, key))
 
-    def pool_is_stalled(self):
-        """Return True if no active, queued or clock trigger awaiting tasks"""
+    def is_stalled(self):
+        """Return True if the suite is stalled.
+
+        A suite is stalled when:
+        * It has no active tasks.
+        * It has waiting tasks with unmet prerequisites
+          (ignoring clock triggers).
+        """
         can_be_stalled = False
         for itask in self.get_tasks():
-            if itask.point > self.stop_point:
-                # Don't consider task beyond stop point
-                continue
-            if itask.state.status in [
+            if itask.point > self.stop_point or itask.state.status in [
                     TASK_STATUS_SUCCEEDED, TASK_STATUS_EXPIRED]:
-                # Succeeded and expired tasks don't stall the suite
+                # Ignore: Task beyond stop point.
+                # Ignore: Succeeded and expired tasks.
                 continue
             if itask.state.status in TASK_STATUSES_NOT_STALLED or (
-                    not itask.start_time_reached() and
-                    itask.state.status not in TASK_STATUSES_FINAL):
+                    itask.state.status in TASK_STATUS_HELD and
+                    itask.state.hold_swap in TASK_STATUSES_NOT_STALLED):
+                # Pool contains active tasks (or held active tasks)
+                # Return "not stalled" immediately.
                 return False
+            if ((itask.state.status == TASK_STATUS_WAITING or
+                    itask.state.hold_swap == TASK_STATUS_WAITING) and
+                    itask.state.prerequisites_are_all_satisfied()):
```

Comment:
Based on the below comments and the purpose of this check, can you tighten up the logic slightly so it'll only carry out the prereq_are_all_satisfied() part of the check for tasks which are clock triggered? We've seen in the cases where there are large numbers of family triggers going into a particular task that cpu can spike on prereq checking so we should only perform the check when needed.